File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.js
class Solution {
    topKFrequent(nums, k) {
        let result = {}
        nums.forEach((item)=>{
            if(result[item]){
                result[item]=result[item]+1
            }else{
                result[item]=1
            }
        })
        let arr = new Array(nums.length+1).fill([])
        let keys= Object.keys(result)
        let values= Object.values(result)
        values.forEach((item,index)=>{
            arr[item]=[...arr[item],keys[index]]
        })
        let kmostresult=[]
        for(let i=arr.length-1;i>=0 && k!=0;i--){
            if(arr[i].length){
                kmostresult=[...kmostresult,...arr[i]]
                k-=arr[i].length
            }
        }
        return kmostresult
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        L,R,result=0,len(nums)-1,nums[0]
        if nums[L]<=nums[R]:
          return result
        while L<=R: 
          mid=(L+R)//2
          if nums[L]<=nums[mid]:
            result=min(result,nums[L])
            L=mid+1
          elif nums[mid]<nums[L]:
            R=mid
        return result
